build.gradle.kts

```diff
@@ -156,147 +161,146 @@ subprojects {
+        minimize() {
+            // Sometimes minimisation takes away too much ... ensure we keep these.
+            exclude(dependency("com.fasterxml.jackson.core:.*:.*"))
+            exclude(dependency("org.commonjava.maven.ext:.*:.*"))
+            exclude(dependency("org.commonjava.maven.atlas:.*:.*"))
+            exclude(dependency("org.aeonbits.owner:.*:.*"))
+            exclude(dependency("org.slf4j:.*:.*"))
+            exclude(dependency("org.apache.maven:.*:.*"))
+        }
+        doFirst {
+            manifest {
+                attributes["Built-By"] = System.getProperty("user.name")
+                attributes["Build-Timestamp"] = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(Date())
+                attributes["Scm-Revision"] = Grgit.open(mapOf("currentDir" to project.rootDir)).use { g -> g.head().id }
+                attributes["Created-By"] = "Gradle ${gradle.gradleVersion}"
+                attributes["Build-Jdk"] = System.getProperty("java.version") + " ; " + System.getProperty("java.vendor") + " ; " + System.getProperty("java.vm.version")
+                attributes["Build-OS"] = System.getProperty("os.name") + " ; " + System.getProperty("os.arch") + " ; " + System.getProperty("os.version")
+                attributes["Implementation-Version"] = "${project.version}"
             }
         }
+    }
 
-        val sourcesJar by tasks.registering(Jar::class) {
-            archiveClassifier.set("sources")
-            from(sourceSets.main.get().allSource)
-        }
+    val sourcesJar by tasks.registering(Jar::class) {
+        archiveClassifier.set("sources")
+        from(sourceSets.main.get().allSource)
+    }
 
-        val javadocJar by tasks.registering(Jar::class) {
-            archiveClassifier.set("javadoc")
-            from(tasks["javadoc"])
-        }
+    val javadocJar by tasks.registering(Jar::class) {
+        archiveClassifier.set("javadoc")
+        from(tasks["javadoc"])
+    }
 
-        // configure publishing of the shadowJar
-        configure<PublishingExtension> {
-            publications {
-                create<MavenPublication>("shadow") {
-                    project.shadow.component(this)
-                    artifact(sourcesJar.get())
-                    artifact(javadocJar.get())
-
-                    // we publish the init gradle file to make it easy for tools that use
-                    // the plugin to set it up without having to create their own init gradle file
-                    if (project.name == "analyzer") {
-                        artifact("${sourceSets.main.get().output.resourcesDir}/analyzer-init.gradle") {
-                            classifier = "init"
-                            extension = "gradle"
-                        }
+
+    // configure publishing of the shadowJar
+    configure<PublishingExtension> {
+        publications {
+            create<MavenPublication>("shadow") {
```

Comment:
As per above re comment for shadow - I think a different publication will be needed for common versus cli/analyzer/etc?